.checkstyle/checkstyle.xml

```diff
@@ -92,7 +92,7 @@
         </module>
         <module name="ClassDataAbstractionCoupling">
             <!-- default is 7 -->
-            <property name="max" value="20"/>
+            <property name="max" value="21"/>
```

Comment:
Out of interest, which class breached the existing limit? It may be better to refactor than just bump the limit.

Reply:
It was the `ControllerTest` class.

```
[ERROR] /home/ppatiern/github/strimzi/topic-controller/src/test/java/io/strimzi/controller/topic/ControllerTest.java:39:1: Class Data Abstraction Coupling is 21 (max allowed is 20) classes [AtomicInteger, ClusterAuthorizationException, Config, ConfigEntry, ConfigMapBuilder, Controller, KubernetesClientException, MockK8s, MockKafka, MockTopicStore, Node, Topic.Builder, TopicDescription, TopicExistsException, TopicMetadata, TopicName, TopicPartitionInfo, TopicStore.EntityExistsException, TopicStore.NoSuchEntityExistsException, UnknownTopicOrPartitionException, io.strimzi.controller.topic.Config]. [ClassDataAbstractionCoupling]
```

Reply:
`ClassDataAbstractionCoupling` is basically (iiuc) a count of the number of classes which are instantiated in a class. It's a rather arbitrary thing to measure, but the more classes something instantiates the more coupled it is with those things. I wonder if there are some utility methods we could factor out of the ControllerTest to reduce this number? For example the creation of the topic metadata (`getTopicMetadata()`) seems to be be something that's also done in the `TopicSerializationTest` and the `MockAdminClient`.

Reply:
@tombentley why the current checkstyle says that default is 7 ? did you increase it to 20 ?

Reply:
I don't see it used by `MockAdminClient` but for sure we can do that factoring it out as static method in an Utility class. For now we should at least passing the topic name as parameter (all the others seem to be the same).